### PR TITLE
fix: support updating domain, endpoint path and auth secret of ingestion server

### PR DESCRIPTION
**Pull request by dengmingtong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

Support updating domain name, endpoint path and auth secret arn

## Implementation highlights

1, Support updating 
  - domain name
  - endpoint path
  - auth secret arn

2, Add unit test cases for above changes

## Test below cases:

1, third sdk + http
  - change endpoint path
  - add appId
  - change endpoint path again

2, third sdk + https + disable auth
  - change endpoint path
  - add appId
  - change endpoint path again

3, third sdk, https, enable auth
  - change auth arn
  - change domain
  - change endpoint path
  - change above together
  - add appId

4, clickstream sdk + http
  - change endpoint path
  - add appId
  - change endpoint path again
  - add and remove appId

5, clickstream sdk + https + disable auth
  - change endpoint path
  - change domain name
  - add appId
  - change endpoint path and domain name again
  - add and remove appId

6, clickstream sdk + https + enable auth
  - change endpoint path, domain name and auth arn
  - add appIds
  - change endpoint path, domain name and auth arn again
  - remove one of appId

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/ingestion-server/custom-resource/update-alb-rules/index.ts [hunk 1]
```diff
@@ -55,6 +55,7 @@ type ResourceEvent = CloudFormationCustomResourceEvent;
 
 export const handler = async (event: ResourceEvent, context: Context) => {
   try {
+    logger.info('input event: ', { event });
```
Comment (zxkane) [inline src/ingestion-server/custom-resource/update-alb-rules/index.ts:58]: It's a redundancy log. We enabled `POWERTOOLS_LOGGER_LOG_EVENT: 'true',` to log the event of Lambda.

File: test/ingestion-server/custom-resource/update-alb-rules.test.ts [hunk 2]
```diff
@@ -871,134 +1210,78 @@ test('Update Listener for ClickStream SDK, with HTTP, with appIds, and have exis
...
       ...basicEvent.ResourceProperties,
-      appIds: 'notepad1,notepad2',
-      endpointPath: '/collectnew',
-      clickStreamSDK: 'YES',
-      authenticationSecretArn: 'arn:aws:secretsmanager:us-east-1:11111111111:secret:test-secret',
+      appIds: 'notepad1,notepad2,nodePad3',
+      clickStreamSDK: 'Yes',
+      authenticationSecretArn: '',
       protocol: 'HTTP',
+      authenticationSecretArn: '',
```
Comment (zxkane) [inline test/ingestion-server/custom-resource/update-alb-rules.test.ts:1236]: duplicated to line 1234

File: test/ingestion-server/custom-resource/update-alb-rules.test.ts [hunk 3]
```diff
@@ -871,134 +1210,78 @@ test('Update Listener for ClickStream SDK, with HTTP, with appIds, and have exis
...
+    },
+    OldResourceProperties: {
```
Comment (zxkane) [inline test/ingestion-server/custom-resource/update-alb-rules.test.ts:1238]: The old properties do not have `appIds`. So will the case work for updating existing rules with the application ID?

File: src/ingestion-server/custom-resource/update-alb-rules/index.ts (hunk 1 above)
Comment (dengmingtong) [reply src/ingestion-server/custom-resource/update-alb-rules/index.ts:58]: Removed this line

File: test/ingestion-server/custom-resource/update-alb-rules.test.ts (hunk 2 above)
Comment (dengmingtong) [reply test/ingestion-server/custom-resource/update-alb-rules.test.ts:1236]: Yes, removed it.

File: test/ingestion-server/custom-resource/update-alb-rules.test.ts (hunk 3 above)
Comment (dengmingtong) [reply test/ingestion-server/custom-resource/update-alb-rules.test.ts:1238]: We do not use appIds in old properties, just use appIds in new properties to compare with existing rules and update with new appIds.
In this cases, the existing rules contains nodePad3 and nodePad4, new appIds is notepad1,notepad2,nodePad3.
The notepad1,notepad2 rules will be added, and nodePad4 rule will be deleted.
The below code is checking this:
```
  expect(albClientMock).toHaveReceivedNthCommandWith(9, DeleteRuleCommand, {
    RuleArn: 'RuleArn5',
  });
```

**Review by zxkane — APPROVED** [review]

**Pull request merged**